libnd4j/include/ops/declarable/generic/nn/batchnorm.cpp

```diff
@@ -29,82 +29,6 @@ namespace nd4j {
 namespace ops {
 
 
-CUSTOM_OP_IMPL(batchnorm, 3, 1, false, 1, 2) {    
-    auto input    = INPUT_VARIABLE(0);
-    auto mean     = INPUT_VARIABLE(1);
-    auto variance = INPUT_VARIABLE(2);
-    NDArray *gamma    = nullptr;
-    NDArray *beta     = nullptr;
-
-    auto output   = OUTPUT_VARIABLE(0);
-
-    const bool applyScale  = (bool)INT_ARG(0);
-    const bool applyOffset = (bool)INT_ARG(1);
-
-    // FIXME: double?
-    const double epsilon     = T_ARG(0);
-
-    if(applyScale)
-        gamma = INPUT_VARIABLE(3);    
-    if(applyOffset)
-        beta = INPUT_VARIABLE(3 + static_cast<int>(applyScale));    
-
-    std::vector<const NDArray*> inArrs(block.width());
-    for(int i = 0; i < block.width(); ++i)
-        inArrs[i] = INPUT_VARIABLE(i);
-
-    // check whether all input shapes are mutually broadcastable
-    Nd4jLong* outShapeInfo = nullptr;
-    const bool areShapesOk = ShapeUtils::evalCommonBroadcastShapeInfo(inArrs, outShapeInfo, block.getWorkspace());
-    REQUIRE_TRUE(areShapesOk, 0, "BATCHNORM op: the shapes of input arrays are not mutually broadcastable !");
-
-    // normalized output = gamma * ((input - mean) / sqrt(variance + epsilon)) + beta
-
-    auto sigmaInvGam = (*variance + epsilon).transform(transform::RSqrt);
-    if(applyScale)
-        sigmaInvGam *= *gamma;
-
-    NDArray inputMinusMean;
-    if(!input->isSameShape(output) && !mean->isSameShape(output)) {
-        auto inputTiled = NDArray(output, false, block.launchContext());
-        input->tile(inputTiled);
-        inputMinusMean = inputTiled - *mean;
-    }
-    else
-        inputMinusMean = *input - *mean;       
-
-    if (applyOffset)
-        output->assign(inputMinusMean * sigmaInvGam + *beta);
-    else 
-        output->assign(inputMinusMean * sigmaInvGam);
-
-    return Status::OK();
-}
-
-    DECLARE_TYPES(batchnorm) {
-        getOpDescriptor()
-                ->setAllowedInputTypes(nd4j::DataType::ANY)
-                ->setAllowedOutputTypes({ALL_FLOATS});
-    }
-
-
-//////////////////////////////////////////////////////////////////////////
-DECLARE_SHAPE_FN(batchnorm) {        
-
-    std::vector<const NDArray*> inArrs(block.width());
-    auto in = inputShape->at(0);
-    for(int i = 0; i < block.width(); ++i)
-        inArrs[i] = INPUT_VARIABLE(i);
-
-    // check whether all input shapes are mutually broadcastable
-    Nd4jLong* outShapeInfo = nullptr;
-    const bool areShapesOk = ShapeUtils::evalCommonBroadcastShapeInfo(inArrs, outShapeInfo, block.getWorkspace());
-    REQUIRE_TRUE(areShapesOk, 0, "BATCHNORM op: the shapes of input arrays are not mutually broadcastable !");
-
-    auto result = ConstantShapeHelper::getInstance()->createShapeInfo(ShapeDescriptor(outShapeInfo, DataTypeUtils::pickFloatingType(ArrayOptions::dataType(in))));
-    return SHAPELIST(result);
-}
-
 //////////////////////////////////////////////////////////////////////////
 CUSTOM_OP_IMPL(batchnorm_new, 3, 1, false, 1, 2) {
```

Comment:
since you had removed `batchnorm` - let's rename `batchnorm_new` to `batchnorm` now. 

p.s. and update mkldnn platform helper then. and java wrapper.